### PR TITLE
[SPARK-30514][K8S] add python environment support for JavaMainAppResource

### PR DESCRIPTION
**Pull request by jackylee-ch**
### What changes were proposed in this pull request?
If application is started with JavaMainAppResource, no python environment, such as PYSPARK_MAJOR_PYTHON_VERSION and PYSPARK_EXECUTOR_MEMORY, won't be available in Executor Pods. It make it not easy to work with Livy. In apache Livy, the program is first started with JavaMainAppResource, and then start the Python worker. At this time, the program needs to be able to pass Python environment variables.

In spark on yarn, we support this through spark.yarn.isPython. In k8s, we should also support this.

### Why are the changes needed?
To support Livy on spark with k8s, we need this config.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Add new unit tests and run all unit tests

### CODE REVIEW

**Comment by stczwd** [comment]
cc @vanzin  @srowen

**Comment by stczwd** [comment]
cc @srowen @dongjoon-hyun @holdenk

**Comment by dongjoon-hyun** [comment]
ok to test

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverCommandFeatureStep.scala [hunk 1]
```diff
@@ -62,7 +62,18 @@ private[spark] class DriverCommandFeatureStep(conf: KubernetesDriverConf)
...
+        .withName(ENV_PYSPARK_MAJOR_PYTHON_VERSION)
+        .withValue(conf.get(PYSPARK_MAJOR_PYTHON_VERSION))
+        .build())
+    } else {
+      Seq[EnvVar]()
+    }
+
+    val driverContainer = baseDriverContainer(pod, res)
+      .addAllToEnv(javaEnvs.asJava)
+      .build()
```
Comment (dongjoon-hyun) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverCommandFeatureStep.scala]: Hi, @stczwd . This PR duplicates the code here and in `configureForPython`. Instead, can we refactor this duplicated logic into a common function?

Comment (stczwd) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverCommandFeatureStep.scala]: Yeap, glad to do that. Thanks for reply.

**Pull request closed without merging**